pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -108,33 +93,6 @@ func (i *instantSplitter) copyWithEmbeddedExpr(embeddedExpr *parser.AggregateExp
 	return &instantSplitter
 }
 
-// isSplittable returns whether it is possible to optimize the given sample expression.
-func isSplittable(expr parser.Expr) bool {
-	switch e := expr.(type) {
-	case *parser.AggregateExpr:
-		// A vector aggregation is splittable, if the aggregation operation is supported and the inner expression is also splittable.
-		return supportedVectorAggregators[e.Op] && isSplittable(e.Expr)
```

Comment:
I'm not sure if this will introduce problems 🤔 (need to think a bit more about this).
But by removing this we are opening the door to all expressions wrapped by vector aggregators to be split.
Maybe it won't be a problem because we only send downstream the `splittableVectorAggregators` vectors 🕵️‍♀️

Reply:
Ok, let's investigate if this could cause issues. If it does, then we should port this logic to `instantSplitter.mapAggregatorExpr()`.

In `instantSplitter.mapAggregatorExpr()` we have:
https://github.com/grafana/mimir/blob/cc0ef427c9c2e516d844479028c2c15a726be740/pkg/frontend/querymiddleware/astmapper/instant_splitting.go#L149-L153

Since `splittableVectorAggregators` is already stricter, the risk of this change could from here:
https://github.com/grafana/mimir/blob/cc0ef427c9c2e516d844479028c2c15a726be740/pkg/frontend/querymiddleware/astmapper/instant_splitting.go#L152